modin/engines/base/io/text/text_file_dispatcher.py

```diff
@@ -232,29 +286,165 @@ def _read_rows(
+            The maximum number of bytes, that can be read during function call.
+        from_begin: bool, default False
+            Whether or not to move file pointer to the beginning of the file.
+        save_position: bool, default False
+            Whether or not to restore file pointer position after rows reading.
+        skiprows: array or callable, default None
+            Specifies rows to skip.
 
         Returns
         -------
-        tuple of bool and int,
+        tuple of bool and two ints,
             bool: If file pointer reached the end of the file, but did not find
                 closing quote returns `False`. `True` in any other case.
-            int: Number of rows that was read.
+            int: Number of rows that was read (including skipped).
+            int: Number of rows that was "considered" (read rows excluding skipped)
         """
+        if nrows is None and max_bytes is None:
+            max_bytes = float("inf")
+
         if nrows is not None and nrows <= 0:
-            return True, 0
+            return True, 0, 0
+
+        # we need this condition to avoid unnecessary checks in `stop_condition`
+        # which executes in a huge for loop
+        if nrows is not None and max_bytes is None:
+            stop_condition = lambda rows_readed: rows_readed >= nrows  # noqa (E731)
+        elif nrows is not None and max_bytes is not None:
+            stop_condition = (
+                lambda rows_readed: f.tell() >= max_bytes or rows_readed >= nrows
+            )  # noqa (E731)
+        else:
+            stop_condition = lambda rows_readed: f.tell() >= max_bytes  # noqa (E731)
+
+        if max_bytes is not None:
+            max_bytes = max_bytes + f.tell()
+
+        start_position = f.tell()
+        if from_begin:
+            f.seek(0, os.SEEK_SET)
+
+        rows_considered = 0
+        rows_readed = 0
 
-        rows_read = 0
+        should_handle_skiprows = skiprows is not None and not isinstance(skiprows, int)
+
+        def skiprows_handler_builder(skiprows):
+            if callable(skiprows):
+
+                def stepper():
+                    row_number = 0
+                    while True:
+                        yield not skiprows(row_number)
+                        row_number += 1
+
+            elif is_list_like(skiprows):
+
+                def stepper():
+                    row_number = 0
+                    index_to_compare = 0
+                    while index_to_compare < len(skiprows):
+                        if skiprows[index_to_compare] == row_number:
+                            index_to_compare += 1
+                            yield 0
+                        else:
+                            yield 1
+                        row_number += 1
+                    while True:
+                        yield 1
+
+            else:
+
+                def stepper():
+                    while True:
+                        yield 1
```

Comment:
that huge dispatching probably should be moved to a separate function

Reply:
please also add a comment about how `stepper` works for list-like skiprows

Reply:
Separate function and comment are added.